File: app/src/main/java/tech/michaeloverman/mscount/pojos/Click.java
/* Copyright (C) 2017 Michael Overman - All Rights Reserved */
package tech.michaeloverman.mscount.pojos;

/**
 * POJO to handle aspects of the sound files for clicks
 * Created by Michael on 10/6/2016.
 */

public class Click {
    private final String mAssetPath;
    private final String mName;
    private Integer mSoundId;

    public Integer getSoundId() {
        return mSoundId;
    }

    public void setSoundId(Integer soundId) {
        mSoundId = soundId;
    }

    public Click(String assetPath) {
        mAssetPath = assetPath;
        String[] components = assetPath.split("/");
        String filename = components[components.length - 1];
        mName = filename.replace(".ogg", "");
    }

    public String getAssetPath() {
        return mAssetPath;
    }

    public String getName() {
        return mName;
    }
}

File: app/src/main/java/tech/michaeloverman/mscount/pojos/Click.kt
/* Copyright (C) 2017 Michael Overman - All Rights Reserved */
package tech.michaeloverman.mscount.pojos

/**
 * POJO to handle aspects of the sound files for clicks
 * Created by Michael on 10/6/2016.
 */
class Click(val assetPath: String) {
    val name: String
    var soundId: Int = -1

    init {
        val components = assetPath.split("/").toTypedArray()
        val filename = components[components.size - 1]
        name = filename.replace(".ogg", "")
    }
}